pylearn2/utils/data_specs.py

```diff
@@ -265,7 +266,7 @@ def nest(self, flat):
                 assert self.n_unique_specs == 1
             return self._make_nested_space(flat, self.spec_mapping)
         else:
-            if isinstance(flat, tuple):
+            if isinstance(flat, Sized):
```

Comment:
Yes, but you would need to convert the list to a tuple before calling `_make_nested_tuple`, or update `_make_nested_tuple` in the same way.